database/skylink.go

```diff
@@ -0,0 +1,99 @@
+package database
+
+import (
+	"context"
+
+	accdb "github.com/SkynetLabs/skynet-accounts/database"
+	"gitlab.com/NebulousLabs/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+var (
+	// ErrInvalidSkylink is returned when a client call supplies an invalid
+	// skylink hash.
+	ErrInvalidSkylink = errors.New("invalid skylink")
+	// ErrSkylinkExists is returned when we try to create a skylink that already
+	// exists.
+	ErrSkylinkExists = errors.New("skylink already exists")
+	// ErrSkylinkNoExist is returned when we try to get a skylink that doesn't
+	// exist.
+	ErrSkylinkNoExist = errors.New("skylink does not exist")
+)
+
+type (
+	// Skylink represents a skylink object in the DB.
+	Skylink struct {
+		ID      primitive.ObjectID `bson:"_id,omitempty"`
+		Skylink string             `bson:"skylink"`
+		Servers []string           `bson:"servers"`
+	}
+)
+
+// SkylinkCreate inserts a new skylink into the DB. Returns an error if it
+// already exists.
+func (db *DB) SkylinkCreate(ctx context.Context, sl string, server string) (Skylink, error) {
+	if !accdb.ValidSkylinkHash(sl) {
+		return Skylink{}, ErrInvalidSkylink
+	}
+	if server == "" {
+		return Skylink{}, errors.New("invalid server name")
+	}
+	s := Skylink{
+		Skylink: sl,
+		Servers: []string{server},
+	}
+	ir, err := db.staticDB.Collection(collSkylinks).InsertOne(ctx, s)
```

Comment:
wondering whether we should extend the `servers` array if the current server is not in there? so if you have a race condition you end up with a correct view of the skylink object rather than an error on one side? Just a thought.

Reply:
Makes sense, added.